Neos.Flow/Classes/Security/Policy/Roles.php

```diff
@@ -0,0 +1,116 @@
+<?php
+declare(strict_types=1);
+
+namespace Neos\Flow\Security\Policy;
+
+use Neos\Flow\Annotations as Flow;
+
+/*
+* This file is part of the Neos.Flow package.
+*
+* (c) Contributors of the Neos Project - www.neos.io
+*
+* This package is Open Source Software. For the full copyright and license
+* information, please view the LICENSE file which was distributed with this
+* source code.
+*/
+
+/**
+ * A collection class containing all roles for an account
+ *
+ * @Flow\Proxy(false)
+ */
+final class Roles implements \JsonSerializable, \IteratorAggregate, \Countable, \ArrayAccess
```

Comment:
Why do we need the `\ArrayAccess`? With it in place we have to throw the `BadMethodCallException` below and it seems not necessary!?